src/Features/Core/Portable/DocumentSpanExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Navigation;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class DocumentSpanExtensions
+    {
+        public static bool CanNavigateTo(this DocumentSpan documentSpan)
+        {
+            var workspace = documentSpan.Document.Project.Solution.Workspace;
+            var service = workspace.Services.GetService<IDocumentNavigationService>();
+            return service.CanNavigateToSpan(workspace, documentSpan.Document.Id, documentSpan.SourceSpan);
```

Comment:
these look super familiar.  Did we not have these elsewhere?

Reply:
part of this file has moved to other file. looks like git mark that file as moved and this file as new.